modules/ROOT/pages/mtf-testing-sources.adoc

```diff
@@ -0,0 +1,115 @@
+= Sources
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+
+Because you must place sources at the beginning of a flow, they cannot appear directly in the test. To test a source, you must have a flow with the source you want to test, and then a test that triggers the source.
+
+There are several types of sources, each with its own distinct testing strategy.
+
+== Sources That Emit a Response
+
+The most common type of test is when the source generates a synchronous response. To test a source that emits a response, you must trigger the source to get that response, and then assert it.
+
+For example, in the case of `http:listener`, you can trigger the source with `http:request`:
+
+[source, xml, linenums]
+----
+<flow name="responseSourceFlow">
+    <test-connector:listener config-ref="aConfig" />
+    <logger level="DEBUG" message="Listener executed"/>
+</flow>
+
+<munit:test name="responseSourceFlowTest">
+    <munit:enable-flow-sources>
+        <munit:enable-flow-source value="responseSourceFlow"/>
+    </munit:enable-flow-sources>
+    <munit:execution>
+        <!-- Hit the source -->
+        <...>
+    </munit:execution>
+    <munit:validation>
+        <munit-tools:assert-equals actual="#[payload]" expected="#['Example']"/>
+    </munit:validation>
+</munit:test>
+----
+
+== Polling Sources
+
+Because a polling source polls items periodically with a given scheduling strategy, the test must trigger the polling source and wait for the next poll. Then, the test can validate the response generated by the polling source.
+
+=== Trigger a Polling Source a Single Time
+
+You can trigger the polling source a single time. For example, for a `file:listener` operation, you can trigger the polling source by creating new files:
+
+[source, xml, linenums]
+----
+<flow name="pollingSourceFlow">
+    <test-connector:polling-source>
+        <scheduling-strategy>
+            <fixed-frequency frequency="500"/>
+        </scheduling-strategy>
+    </test-connector:polling-source>
+    <munit-tools:queue>
+        <munit-tools:value>#[payload]</munit-tools:value>
+    </munit-tools:queue>
+</flow>
+
+<munit:test name="pollingSourceTest">
+    <munit:enable-flow-sources>
+        <munit:enable-flow-source value="pollingSourceFlow"/>
+    </munit:enable-flow-sources>
+    <munit:execution>
+        <!-- Trigger the source -->
+        <...>
+        <!-- Wait for polling source to trigger -->
+        <munit-tools:dequeue target="storedPayload"/>
+    </munit:execution>
+    <munit:validation>
+        <munit-tools:assert-equals actual="#[vars.storedPayload]" expected="#['Example']"/>
+    </munit:validation>
+</munit:test>
+----
+
+=== Trigger a Polling Source Multiple Times
+
+You can trigger the polling source multiple times:
```

Comment:
Any chance we can explain why would someone want to trigger the polling source one vs multiple times? What's the difference/benefit? If that's possible, you could rewrite the first sentence of both scenarios to say "You can trigger <x> times to <reason/benefit>". Just an idea.